src/main/java/seedu/duke/ModuleList.java

```diff
@@ -0,0 +1,175 @@
+package seedu.duke;
+
+import java.util.ArrayList;
+
+public class ModuleList {
+    public static ArrayList<Module> modules = new ArrayList<>();
+    public static int viewCount;
+    public static int mcsCount;
+
+    /**
+     * Function to add a module to the moduleList if it does not already exist
+     * @param mod The module to be added to the list
+     */
+    public void add(Module mod) {
+        boolean isRepeat = checkRepetition(mod);
+        if (isRepeat) {
+            UI.repetitionMessage(mod.getCourse());
+        } else {
+            int before = modules.size();
+            modules.add(mod);
+            UI.addMessage(mod.getCourse(), mod.getSemesterTaken(), mod.getGrade());
+            int after = modules.size();
+
+            try {
+                assert before + 1 == after : "module list size should be increased by 1 after add";
+            } catch (AssertionError e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * function to check if a given module already exists in the moduleList
+     * @param mod the module that needs to be searched for in the moduleList
+     * @return true if the module already exists. False if it does not. Format: boolean
+     */
+    public boolean checkRepetition(Module mod) {
+        for (Module m : modules) {
+            if (m.getCourse().matches(mod.getCourse())) {
```

Comment:
May be complicated expressions. Consider declaring other variables to store the value of `mod.getCourse()` and changing values of `m.getCourse()`, before using them in the loop.